File: spring-cloud-integrated-demo/product-service/src/test/java/com/github/jccode/springcloud/integrateddemo/product/service/ProductServiceTest.java
package com.github.jccode.springcloud.integrateddemo.product.service;


import com.github.jccode.springcloud.integrateddemo.product.model.Product;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.greaterThan;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.nullValue;

@SpringBootTest
@RunWith(SpringRunner.class)
@MybatisTest
@Import(ProductService.class)
@ActiveProfiles("test")
public class ProductServiceTest {

    @Autowired
    private ProductService productService;

    @Test
    public void mybatisWorks() {

        Product product = new Product();
        product.setName("iphone X");
        product.setPrice(6957);
        product.setStock(0);
        assertThat(product.getId(), is(nullValue()));

        int result = productService.save(product);
        assertThat(result, greaterThan(0));
        assertThat(product.getId(), greaterThan(0));
    }
}

File: spring-cloud-integrated-demo/product-service/src/test/scala/com/github/jccode/springcloud/integrateddemo/product/service/ProductServiceTest.scala
package com.github.jccode.springcloud.integrateddemo.product.service

import com.github.jccode.springcloud.integrateddemo.product.model.Product
import org.junit.Test
import org.junit.runner.RunWith
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest
import org.scalatest.Matchers
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit4.SpringRunner

@SpringBootTest
@RunWith(classOf[SpringRunner])
@MybatisTest
@Import(Array(classOf[ProductService]))
@ActiveProfiles(Array("test"))
class ProductServiceTest extends Matchers {

  @Autowired
  val productService: ProductService = null

  @Test
  def mybatisWorks(): Unit = {
    val product = new Product
    product.setName("iphone X")
    product.setPrice(6957)
    product.setStock(0)
    product.getId should be (null)
    val result = productService.save(product)
    result should be >0
    product.getId.toInt should be >0
  }
}
